### PR TITLE
[docs] Add productivity hacks to stdlib progman

### PR DESCRIPTION
**Pull request by milseman**
Small section about using ninja to more efficiently work on the standard library.

### CODE REVIEW

File: docs/StandardLibraryProgrammersManual.md [hunk 1]
```diff
@@ -170,3 +170,23 @@ The standard library utilizes thread local storage (TLS) to cache expensive comp
...
+
+### Be a Ninja
+
+To *be* a productivity ninja, one must *use* `ninja`. `ninja` can be invoked inside the swift build directory, e.g. `<path>/build/Ninja-ReleaseAssert/swift-macosx-x86_64/`. Running `ninja` (which is equivalent to `ninja all`) will build the local swift, stdlib and overlays. It doesn’t necessarily build all the testing infrastructure, benchmarks, etc.
+
+`ninja -t targets` gives a list of all possible targets to pass to ninja. This is useful for grepping.
+
+For this example, we will figure out how to quickly iterate on a change to the standard library to fix 32-bit build errors while building on a 64-bit host, suppressing warnings along the way.
+
+`ninja -t tagets | grep stdlib | grep i386` will output many targets, but at the bottom we see `swift-stdlib-iphonesimulator-i386`, which looks like a good first step. This target will just build i386 parts and not waste our time also building the 64-bit stdlib, overlays, etc.
```
Comment (xwu) [inline docs/StandardLibraryProgrammersManual.md]: Typo: `tagets` [sic].

**Review by lancep — APPROVED** [review]
Looks good!

**Comment by milseman** [comment]
@swift-ci please smoke test and merge

**Comment by milseman** [comment]
@swift-ci please smoke test

**Pull request merged**